qcodes/instrument/delegate/delegate_instrument.py

```diff
@@ -237,14 +237,15 @@ def _add_parameter(
         name: str,
         source: Parameter,
     ) -> DelegateGroupParameter:
+        param_name = "{group_name}_{name}"
+        if param_name in self.parameters:
+            raise KeyError(f"Duplicate parameter name {param_name} on {self.name}")
         param = self.param_cls(
             name=f"{group_name}_{name}",
             instrument=self,
-            source=source
+            source=source,
+            bind_to_instrument=True,
```

Comment:
For safety this should probably also be wrapped in a try catch with parameters that do not take bind_to_instrument as a kwarg and handle parameters that do not pass instrument correctly

Reply:
This is done now just need to write a test for it

Reply:
rewrote this to use add_parameter to avoid code duplication